packages/jsii-pacmak/lib/targets/kotlin.ts

```diff
@@ -0,0 +1,66 @@
+import spec = require('jsii-spec');
+import { PackageInfo, Target } from '../target';
+import { shell } from '../util';
+import { KotlinGenerator } from './kotlin/kotlingenerator';
+import { getJvmPackageInfos, findJavaRuntimeLocalRepository } from './jvm/mavenutils';
+
+export class Kotlin extends Target {
+  public static toPackageInfos(assm: spec.Assembly): { [language: string]: PackageInfo } {
+    const kotlinTarget = assm.targets && assm.targets.kotlin;
+    const groupId = kotlinTarget && kotlinTarget.maven && kotlinTarget.maven.groupId;
+    const artifactId = kotlinTarget && kotlinTarget.maven && kotlinTarget.maven.artifactId;
+    const version = assm.version;
+    return getJvmPackageInfos('kotlin', groupId, artifactId, version);
```

Comment:
If you're able to assume that the `kotlin` target is configured at this stage (aka it was checked before you got there), you can `assm.targets!.kotlin` instead of using the `x && x.y` madness.